tests/integ/test_packaging.py

```diff
@@ -539,3 +575,258 @@ def plus_one_month(x):
+        "pyyaml",
+        "scipy",
+        "numpy",
+    }
+
+    def import_scikit_fuzzy() -> str:
+        import skfuzzy as fuzz
+
+        return f"{fuzz.__version__}"
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+    session.udf.register(import_scikit_fuzzy, name=udf_name)
+    Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("0.4.2")])
+
+    session.clear_packages()
+    session.clear_imports()
+
+    # Use existing zip file to run the same function again
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        with patch(
+            "snowflake.snowpark.session.Session._upload_unsupported_packages",
+            side_effect=Exception("This function should not have been called"),
+        ):
+            # This should not raise error because we no long call _upload_unsupported_packages (we load it from env)
+            session.add_requirements(
+                test_files.test_unsupported_requirements_file,
+                persist_path=temporary_stage,
+            )
+
+    stage_files = session._list_files_in_stage(temporary_stage)
+    assert f"{zip_file}.gz" in stage_files
+    assert metadata_file in stage_files
+
+    session_imports = session.get_imports()
+    assert len(session_imports) == 1
+    assert f"{temporary_stage}/{zip_file}" in session_imports[0]
+    assert set(session.get_packages().keys()) == {
+        "matplotlib",
+        "pyyaml",
+        "scipy",
+        "numpy",
+    }
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+    session.udf.register(import_scikit_fuzzy, name=udf_name)
+    Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("0.4.2")])
+
+    # Add a second environment
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_packages(["sktime==0.20.0"], persist_path=temporary_stage)
+
+    assert set(session.get_packages().keys()) == {
+        "matplotlib",
+        "pyyaml",
+        "scipy",
+        "numpy",
+        "pandas",
+        "python-dateutil",
+        "scikit-learn",
+        "six",
+        "wrapt",
+    }
+
+    # Assert that metadata contains two environment signatures
+    metadata_path = f"{temporary_stage}/{metadata_file}"
+    metadata = {
+        row[0]: row[1].split("|") if row[1] else []
+        for row in (
+            session.sql(
+                f"SELECT t.$1 as signature, t.$2 as packages from '@{metadata_path}' t"
+            )._internal_collect_with_tag()
+        )
+    }
+    assert len(metadata) == 2
+
+
+# TODO: V2 - Fix this test
+@pytest.mark.xfail(reason="Flaky test")
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC or IS_WINDOWS or not is_pandas_and_numpy_available,
+    reason="Numpy and pandas needed and subprocess process calls might occur (not allowed inside stored proc). "
+    "Also, replicate_local_environment() currently causes an infinite loop in Windows environments.",
+)
```

Comment:
if this is a known flaky test, do you want to hold off on adding it? Do you know what is the cause of flakiness?